tonic/src/status.rs

```diff
@@ -35,7 +35,17 @@ const ENCODING_SET: &AsciiSet = &CONTROLS
 /// assert_eq!(status1.code(), status2.code());
 /// ```
 #[derive(Clone)]
-pub struct Status {
+pub struct Status(Box<StatusInner>);
+
+impl From<StatusInner> for Status {
```

Comment:
This was marginally more ergonomic than creating a constructor, but let me know if you would prefer a constructor instead

Reply:
I think we need to make this a private function on status so that it doesn't show up in the docs? I don't want the docs to even reference `StatusInner` at all.

Reply:
Good point, although `#[doc(hidden)]` would also work